modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/transaction/TransactionStorage.scala

```diff
@@ -85,6 +85,21 @@ class TransactionStorage[F[_]: Sync](
       }
       .void
 
+  def countAllowedForConsensus: F[Int] =
```

Comment:
We should additionally add a validation to L1 consensus. As we are not sure that after checking condition here the transactions will still be there when starting the consensus during L1 consensus. Starting a round as facilitator may pull them before new own consensus starts. Additionally it's probably good idea to change the `DAGBlock` model to not allow adding empty Set of transactions.